src/RouteCollector.php

```diff
@@ -12,7 +12,6 @@
  * @phpstan-import-type ExtraParameters from DataGenerator
  * @phpstan-import-type RoutesForUriGeneration from GenerateUri
  * @phpstan-import-type ParsedRoutes from RouteParser
- * @final
```

Comment:
This is has been added on purpose and we'll make this class final on v3

Reply:
> This is has been added on purpose

If I may ask, for what purpose was it added? Making it final means something like [CastRouteCollector](https://github.com/pmjones/CastRoute/blob/1.x/src/CastRouteCollector.php) cannot capture or modify information along the way, without (essentially) copying the entire RouteCollector class to replace it.

Reply:
The goal is to move the focus of people extending components away from the implementation details and give me flexibility on evolving the lib. 

We can introduce ways to reduce boilerplate in the downstream if that's helpful, I just prefer to favour composition over inheritance for that component.

Reply:
After having tried my hand at it, composition-over-inheritance is difficult to implement with RouteCollector. After creating a decorator around it and modifying addRoute() to automatically add a route name, I realized that addGroup() calls $this internally, which means it will use its own instance methods, and not the decorator instance methods. For example:

```php
    public function __construct(
        protected FastRouteCollector $fastRouteCollector,
    ) {
    }

    /**
     * @inheritdoc
     */
    public function addRoute(string|array $httpMethod, string $route, mixed $handler, array $extraParameters = []): void
    {
        if (is_string($handler)) {
            $extraParameters[self::ROUTE_NAME] ??= $handler;
        }

        $this->fastRouteCollector->addRoute($httpMethod, $route, $handler, $extraParameters);
    }

    public function addGroup(string $prefix, callable $callback): void
    {
        // uses the FastRoute collector addRoute() under the hood,
        // not the decorator addRoute() method.
        $this->fastRouteCollector->addGroup($prefix, $callback);
    }
```

I think the easiest solution here still is to make RouteCollector non-final and allow extension.